scripts/noaa/gpcc_spi/gpcc_spi_aggregation.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+import json
+from typing import List
+import logging
+from datetime import datetime
+import os
+import csv
+import glob
+from pathlib import Path
+
+from absl import flags
+from absl import app
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('in_pattern', '/tmp/gpcc_spi/out/*.csv',
+                    'The directory the csvs are.')
```

Comment:
What are the CSVs?  Wonder why not have the same program generate both grids and aggregations?

So its easier to run them as a single cron.

Reply:
Was hoping to isolate csv preprocessing and aggregation each as isolated tasks. Isolated tasks are easier to orchestrate and task level retry is easier to implement. 

Other datasets may have several layers of preprocessing and having everything in the same script might not generalize well when we need to refactor.

To your point of running it as a single cron, I added a wrapper script which does that for now. It's just slower to do them in series but as long as we only need to process small amounts of datas in refresh, speed perhaps wouldn't be a limiting factor.